src/lib.rs

```diff
@@ -1877,6 +1877,13 @@ impl Connection {
                     // TODO: in the future we might want to buffer other types
                     // of undecryptable packets as well.
                     let pkt_len = b.off() + payload_len;
+
+                    // Don't add to the queue if this packet is not decryptable
+                    // because of the mismatched size.
+                    if b.len() < pkt_len {
+                        return Err(Error::CryptoFail);
```

Comment:
This needs to use `drop_pkt_on_err()` rather than returning an error directly. Also, `CryptoFail` is probably the wrong error since this has nothing to do with crypto. Should be `InvalidPacket` IMO.

Reply:
Will do with the next comment.